src/libopensc/card-openpgp.c

```diff
@@ -2979,11 +2977,18 @@ pgp_build_extended_header_list(sc_card_t *card, sc_cardctl_openpgp_keystore_info
 		componentnames[0] = "private key";
 		comp_to_add = 1;
 
-		/* TODO ECC import with public key, if necessary as denoted in algorithm caps*/
+		/* import public key as well */
+		if (key_info->u.ec.keyformat == SC_OPENPGP_KEYFORMAT_EC_STDPUB){
+			components[1] = key_info->u.ec.ecpointQ;
+			componentlens[1] = key_info->u.ec.ecpointQ_len;
+			componenttags[1] = 0x99;
+			componentnames[1] = "public key";
+			comp_to_add = 2;
+		}
 
 		/* validate */
-		if ((key_info->u.ec.ecpoint == NULL || key_info->u.ec.ecpoint_len == 0)){
-			sc_log(ctx, "Error: ecpoint required!");
+		if ((key_info->u.ec.ecpointQ == NULL || key_info->u.ec.ecpointQ_len == 0)){
```

Comment:
clang-format suggested the following change:


```
-		if ((key_info->u.ec.ecpointQ == NULL || key_info->u.ec.ecpointQ_len == 0)){
+		if ((key_info->u.ec.ecpointQ == NULL || key_info->u.ec.ecpointQ_len == 0)) {
```

<!-- identifier: clang-format -->
<!-- File: src/libopensc/card-openpgp.c -->